src/internal/components/pluggableVisualizations/pivotTable/widthItemsHelpers.ts

```diff
@@ -0,0 +1,168 @@
+// (C) 2020 GoodData Corporation
+import includes = require("lodash/includes");
+
+import {
+    ColumnWidthItem,
+    IMeasureColumnWidthItem,
+    IAttributeColumnWidthItem,
+    isAttributeColumnWidthItem,
+    isMeasureLocatorItem,
+    isMeasureColumnWidthItem,
+} from "../../../../interfaces/PivotTable";
+
+import {
+    IAttributeFilter,
+    IBucketFilter,
+    isAttributeFilter,
+    IBucketItem,
+} from "../../../interfaces/Visualization";
+
+const isAttributeColumnWidthItemVisible = (
+    _widthItem: IAttributeColumnWidthItem,
+    _filters: IBucketFilter[],
+): boolean => true;
+
+const isMeasureWidthItemMatchedByFilter = (
+    widthItem: IMeasureColumnWidthItem,
+    filter: IAttributeFilter,
+): boolean =>
+    filter.selectedElements.some(selectedElement =>
+        widthItem.measureColumnWidthItem.locators.some(
+            locator =>
+                !isMeasureLocatorItem(locator) &&
+                locator.attributeLocatorItem.element === selectedElement.uri,
+        ),
+    );
+
+const isMeasureColumnWidthItemVisible = (
+    widthItem: IMeasureColumnWidthItem,
+    filters: IBucketFilter[],
+): boolean =>
+    filters.reduce((isVisible, filter) => {
+        if (isAttributeFilter(filter)) {
+            const shouldBeMatched = !filter.isInverted;
+            return isVisible && shouldBeMatched === isMeasureWidthItemMatchedByFilter(widthItem, filter);
+        }
+        return isVisible;
+    }, true);
+
+const isWidthItemVisible = (widthItem: ColumnWidthItem, filters: IBucketFilter[]): boolean =>
+    isAttributeColumnWidthItem(widthItem)
+        ? isAttributeColumnWidthItemVisible(widthItem, filters)
+        : isMeasureColumnWidthItemVisible(widthItem, filters);
+
+const containsMeasureLocator = (widthItem: IMeasureColumnWidthItem): boolean =>
+    widthItem.measureColumnWidthItem.locators.some(locator => isMeasureLocatorItem(locator));
+
+const widthItemLocatorsHaveProperLength = (
+    widthItem: IMeasureColumnWidthItem,
+    measuresCount: number,
+    columnAttributesCount: number,
+): boolean => {
+    return (
+        widthItem.measureColumnWidthItem.locators.length > 0 &&
```

Comment:
Please consider separating this to named variables for better readability (use your own names, these I put below are illustrative and may not reflect the reality)
```
const hasMeasureLocators = widthItem.measureColumnWidthItem.locators.length > 0;
const measureLocatorMatchesColumnsLength = ...
const noMeasuresColumnLength = ...
const columnLengthMatches = measureLocatorMatchesColumnsLength || noMeasuresColumnLength;
return hasMeasureLocators && columnsLengthMatches;
```

Reply:
Done